kew.js

```diff
@@ -299,6 +299,38 @@ Promise.prototype.failBound = function (onFail, scope, var_args) {
 }
 
 /**
+ * Spread a promises outputs to the functions arguments.
+ * @param {?function(this:void, ...): RESULT|undefined} onSuccess
+ * @return {!Promise.<RESULT>} returns a new promise with the output of the onSuccess or
+ *     onFail handler
+ * @template RESULT
+ */
+Promise.prototype.spread = function (onSuccess) {
+  return this
+  .then(allInternal)
```

Comment:
Internally, we're using `setImmediate` for our nextTick function.  Internally, when we call `setImmediate`, Node puts the function on a queue.  Each time we cycle through the event loop, Node pops a single function off the queue.

If we front-load each item in the array onto the queue, we're going to execute them all on sequential executions of the event loop.

A more fair way of implementing this would be to have `allInternal` run in an asynchronous loop, where we execute `onSuccess` for the first element, and then call `setImmediate` to enqueue our next iteration through the array.

This isn't an issue for small arrays, but it could lead to more difficult to predict behavior for larger arrays.